pkg/storage/accessibility_request.go

```diff
@@ -115,11 +115,82 @@ func (s *Store) FetchAccessibilityRequests(ctx context.Context) ([]models.Access
 }
 
 // DeleteAccessibilityRequest marks an accessibility request as deleted
-func (s *Store) DeleteAccessibilityRequest(ctx context.Context, id uuid.UUID, reason models.AccessibilityRequestDeletionReason) error {
+func (s *Store) DeleteAccessibilityRequest(_ context.Context, id uuid.UUID, reason models.AccessibilityRequestDeletionReason) error {
 	const archiveAccessibilityRequestSQL = `UPDATE accessibility_requests
 		SET deleted_at = $1, deletion_reason = $2
 		WHERE id = $3
 `
 	_, err := s.db.Exec(archiveAccessibilityRequestSQL, time.Now().UTC(), reason, id)
 	return err
 }
+
+// CreateAccessibilityRequestAndInitialStatusRecord creates the new request and sets the status to OPEN
+func (s *Store) CreateAccessibilityRequestAndInitialStatusRecord(ctx context.Context, request *models.AccessibilityRequest) (*models.AccessibilityRequest, error) {
+	createdRequest, err := s.CreateAccessibilityRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = s.CreateAccessibilityRequestStatusRecord(ctx, &models.AccessibilityRequestStatusRecord{
+		RequestID: createdRequest.ID,
+		Status:    models.AccessibilityRequestStatusOpen,
+		EUAUserID: createdRequest.EUAUserID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return createdRequest, nil
+}
+
+// FetchAccessibilityRequestMetrics gets a metrics digest for 508 requests
+func (s *Store) FetchAccessibilityRequestMetrics(_ context.Context, startTime time.Time, endTime time.Time) (models.AccessibilityRequestMetrics, error) {
+	type createdQueryResponse struct {
+		CreatedAndOpenCount          int `db:"created_and_open_count"`
+		CreatedAndClosedCount        int `db:"created_and_closed_count"`
+		CreatedAndInRemediationCount int `db:"created_and_in_remediation_count"`
+	}
+	const startedCountSQL = `
+		WITH "started" AS (
+		    SELECT *
+		    FROM accessibility_requests
+		    LEFT JOIN
+		    (
+		        SELECT arsr.request_id, status
+		        FROM accessibility_request_status_records as arsr
+		        INNER JOIN (
+		        	SELECT request_id, max(created_at) as latest
+		        	FROM accessibility_request_status_records
+		        	GROUP BY request_id
+		        ) most_recent
+		        ON arsr.request_id = most_recent.request_id
+		        WHERE arsr.created_at = most_recent.latest
+	        ) most_recent_status_record
+		    ON most_recent_status_record.request_id = accessibility_requests.id
+		    WHERE accessibility_requests.created_at >=  $1
+		      AND accessibility_requests.created_at < $2
+		)
+		SELECT sum(CASE WHEN status = 'OPEN' then 1 end) AS created_and_open_count,
+			   sum(CASE WHEN status = 'CLOSED' then 1 end) AS created_and_closed_count,
+			   sum(CASE WHEN status = 'IN_REMEDIATION' then 1 end) AS created_and_in_remediation_count
+		FROM started
+	`
```

Comment:
I think it's possible to get this data with fewer JOINs and less SQL by using `DISTINCT` and `FILTER`:

```sql
SELECT
 COUNT(*) FILTER (WHERE status = 'OPEN') AS created_and_open_count,
 COUNT(*) FILTER (WHERE status = 'CLOSED') AS created_and_closed_count,
 COUNT(*) FILTER (WHERE status = 'IN_REMEDIATION') AS created_and_in_remediation_count
 FROM (
	SELECT DISTINCT ON (request_id) request_id, status, created_at FROM accessibility_request_status_records
	WHERE request_id IN (
		SELECT id FROM accessibility_requests
		WHERE created_at >= $1 AND created_at < $2
	)
	ORDER BY request_id, created_at DESC
) requests;
```

I find this a little easier to follow (innermost subquery gets IDs of accessibility requests in the time range, next one gets the latest status for each of those requests, and the outermost one counts how many of each status). WDYT?

Side note: I think we're going to want to add indexes to `accessibility_request_status_records` on `request_id` since we're using that query for conditions a lot.

Reply:
I like your implementation. Let me commit it